### PR TITLE
Ssl patch

### PR DESCRIPTION
**Pull request by geekspeed**
Patched to add command line options for setting up ssl connections to Elasticsearch

### CODE REVIEW

File: es2csv.py [hunk 1]
```diff
@@ -267,6 +267,11 @@ def main():
     p.add_argument('-e', '--meta_fields', dest='meta_fields', action='store_true', help='Add meta-fields in output.')
     p.add_argument('-v', '--version', action='version', version='%(prog)s ' + __version__, help='Show version and exit.')
     p.add_argument('--debug', dest='debug_mode', action='store_true', help='Debug mode on.')
+    p.add_argument('--verify-certs', dest='verify_certs', action='store_true', help='Verify SSL Certs (default is False)')
+    p.add_argument('--ca-certs', dest='ca_certs', default='', type=str, help='Location of CA cert file for Cert verification')
+    p.add_argument('--client-cert', dest='client_cert', default='', type=str, help='Location of Client Auth cert')
+    p.add_argument('--client-key', dest='client_key', default='', type=str, help='Location of Client Cert Key')
+    p.add_argument('--use-ssl', dest='use_ssl', action='store_true', help='Use SSL in the Elasticsearch connection')
```
Comment (taraslayshchuk) [inline es2csv.py]: As we are using url it automatically switched to true when you pass https url.
https://github.com/elastic/elasticsearch-py/blob/5.0/elasticsearch/client/__init__.py#L46

File: es2csv.py [hunk 2]
```diff
@@ -267,6 +267,11 @@ def main():
     p.add_argument('-e', '--meta_fields', dest='meta_fields', action='store_true', help='Add meta-fields in output.')
     p.add_argument('-v', '--version', action='version', version='%(prog)s ' + __version__, help='Show version and exit.')
     p.add_argument('--debug', dest='debug_mode', action='store_true', help='Debug mode on.')
+    p.add_argument('--verify-certs', dest='verify_certs', action='store_true', help='Verify SSL Certs (default is False)')
+    p.add_argument('--ca-certs', dest='ca_certs', default='', type=str, help='Location of CA cert file for Cert verification')
```
Comment (taraslayshchuk) [inline es2csv.py]: ca_cert, client_cert and client_key by default are None, https://github.com/elastic/elasticsearch-py/blob/5.0/elasticsearch/connection/http_requests.py#L30
It is better to stay with None then to pass empty string

File: es2csv.py [hunk 3]
```diff
@@ -267,6 +267,11 @@ def main():
     p.add_argument('-e', '--meta_fields', dest='meta_fields', action='store_true', help='Add meta-fields in output.')
     p.add_argument('-v', '--version', action='version', version='%(prog)s ' + __version__, help='Show version and exit.')
     p.add_argument('--debug', dest='debug_mode', action='store_true', help='Debug mode on.')
+    p.add_argument('--verify-certs', dest='verify_certs', action='store_true', help='Verify SSL Certs (default is False)')
```
Comment (taraslayshchuk) [inline es2csv.py]: Could you please add examples for  all of these arguments in Readme?

**Comment by taraslayshchuk** [comment]
Thanks for patch.

**Comment by geekspeed** [comment]
Thanks

**Pull request merged**